File: CartyObjC/CartyObjC/ViewController.m
//
//  ViewController.m
//  CartyObjC
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

#import "ViewController.h"
#import <AppTrackingTransparency/AppTrackingTransparency.h>
#import "NativeViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    if (@available(iOS 14, *))
    {
        [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
        }];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"CartySDK";
    // Do any additional setup after loading the view.
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if([segue.identifier isEqualToString:@"template"])
    {
        NativeViewController *nativeViewController = [segue destinationViewController];
        nativeViewController.isTemplate = YES;
    }
    
}

@end

File: CartySwift/CartySwift/ViewController.swift
//
//  ViewController.swift
//  CartySwift
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

import UIKit
import AppTrackingTransparency

class ViewController: UITableViewController
{
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        if #available(iOS 14, *) {
            ATTrackingManager.requestTrackingAuthorization(completionHandler: {status in
                
            })
        }
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if(segue.identifier == "template")
        {
            let nativeViewController:NativeViewController = segue.destination as! NativeViewController;
            nativeViewController.isTemplate = true;
        }
    }
}

